Source/m_menu.c

```diff
@@ -5076,15 +5085,50 @@ boolean M_Responder (event_t* ev)
 	// [FG] reload current level / go to next level
 	if (M_InputActivated(input_menu_nextlevel))
 	{
-		if (demoplayback && singledemo && !demoskip)
+		if (demoplayback && singledemo && !demonext &&
+		    demowarp == -1 && demoskip_tics == -1)
 		{
-			demoskip = true;
+			demonext = true;
 			I_EnableWarp(true);
 			return true;
 		}
 		else if (G_GotoNextLevel(NULL, NULL))
 			return true;
 	}
+
+        if (M_InputActivated(input_demo_skip))
+        {
+          if (demoplayback && singledemo && !demonext &&
+              demowarp == -1 && demoskip_tics == -1)
+          {
+            fastdemo = !fastdemo;
+            I_InitTimer();
```

Comment:
`basetime` gets re-evaluated in `I_InitTimer()`. Is this intended?

Reply:
Yes, it's intended for game speed change compensation.